src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/services/GetOffencesForPersonService.kt

```diff
@@ -4,23 +4,20 @@ import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.hmppsintegrationapi.gateways.NDeliusGateway
 import uk.gov.justice.digital.hmpps.hmppsintegrationapi.gateways.NomisGateway
-import uk.gov.justice.digital.hmpps.hmppsintegrationapi.gateways.PrisonerOffenderSearchGateway
-import uk.gov.justice.digital.hmpps.hmppsintegrationapi.gateways.ProbationOffenderSearchGateway
 import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.Offence
 import uk.gov.justice.digital.hmpps.hmppsintegrationapi.models.Response
 
 @Service
 class GetOffencesForPersonService(
   @Autowired val nomisGateway: NomisGateway,
-  @Autowired val prisonerOffenderSearchGateway: PrisonerOffenderSearchGateway,
-  @Autowired val probationOffenderSearchGateway: ProbationOffenderSearchGateway,
   @Autowired val nDeliusGateway: NDeliusGateway,
+  @Autowired val getPersonService: GetPersonService,
 ) {
   fun execute(pncId: String): Response<List<Offence>> {
-    val responseFromPrisonerOffenderSearch = prisonerOffenderSearchGateway.getPersons(pncId = pncId)
-    val responseFromProbationOffenderSearch = probationOffenderSearchGateway.getPerson(pncId = pncId)
-    val nomisNumber = responseFromPrisonerOffenderSearch.data.firstOrNull()?.identifiers?.nomisNumber
-    val deliusCrn = responseFromProbationOffenderSearch.data?.identifiers?.deliusCrn
+    val personResponse = getPersonService.execute(pncId = pncId)
+    val nomisNumber = personResponse.data["prisonerOffenderSearch"]?.identifiers?.nomisNumber
+    val deliusCrn = personResponse.data["probationOffenderSearch"]?.identifiers?.deliusCrn
```

Comment:
I think using a combination of [our UpstreamApi enum](https://github.com/ministryofjustice/hmpps-integration-api/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsintegrationapi/models/UpstreamApi.kt) and overriding the `toString()` method or adding a new function, we could remove these magic strings in the `GetPersonService` and in this service.

However, I can live with this for now as I imagine we'll be removing these keys at some point for this service - I leave the decision in your capable hands!